### PR TITLE
bugfix - rebuild

### PR DESCRIPTION
**Pull request by mixmix**
This PR:
- separates the setup requirements for reading / writing (unboxers / validators)
- fixes some bugs in how output of status / progress is caculates

:warning: needs new flumedb changes added

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -87,16 +87,14 @@ module.exports = {
...
       var plugs = {}
       var sync = true
-      for(var k in ssb) {
-        if(ssb[k] && isObject(ssb[k]) && isFunction(ssb[k].since)) {
-          plugs[k] = ssb[k].since.value
-          sync = sync && (plugs[k] === ssb.since.value)
-        }
+      for (var name in ssb.views) {
+        plugs[name] = ssb.views[name].since.value
+        sync = sync && (plugs[name] === ssb.since.value)
```
Comment (mixmix) [inline index.js:92]: :+1:  fixes the output of `sbot status`

File: db.js [hunk 2]
```diff
@@ -24,12 +22,10 @@ module.exports = function db (dir, keys, opts) {
...
-    for (var k in db) {
-      if (db[k] && typeof db[k].since === 'function') {
-        n++
-        var c = db[k].since.value
-        current += (Number.isInteger(c) ? c : -1)
-      }
+    for (var name in db.views) {
+      n++
+      var c = db.views[name].since.value
+      current += (Number.isInteger(c) ? c : -1)
```
Comment (mixmix) [inline db.js:28]: :+1: fixes the `sbot progress` output so that it actually shows all views

File: package.json [hunk 3]
```diff
@@ -15,7 +15,7 @@
   },
   "dependencies": {
     "async-write": "^2.1.0",
-    "flumedb": "^2.1.4",
+    "flumedb": "github:flumedb/flumedb",
```
Comment (mixmix) [inline package.json:18]: Needs new fixes in flumedb that are gonna be merged by you Christian

File: create.js [hunk 4]
```diff
@@ -184,6 +150,7 @@ module.exports = function create (path, opts, keys) {
     })
   }
 
+  /* via last index */
   db.getVectorClock = function (_, cb) {
```
Comment (mixmix) [inline create.js:154]: in several files I've collected methods which depend on a particular view into one place, to make them easier to track + maintain

File: indexes/keys.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+var ViewLevel = require('flumeview-level')
+
+module.exports = function () {
+  return ViewLevel(4, (msg) => [ msg.key ])
+}
```
Comment (mixmix) [inline indexes/keys.js:5]: I've pulled all flume indexes into this folder so you can clearly see on what's doing what here

**Comment by christianbundy** [comment]
These changes LGTM, thanks for this patch!

**Pull request merged**